libafl_bolts/src/llmp.rs

```diff
@@ -664,16 +664,10 @@ impl LlmpMsg {
     pub fn in_shmem<SHM: ShMem>(&self, map: &mut LlmpSharedMap<SHM>) -> bool {
         let map_size = map.shmem.as_slice().len();
         let buf_ptr = self.buf.as_ptr();
+        let len = self.buf_len_padded as usize + size_of::<LlmpMsg>();
         unsafe {
-            if buf_ptr > (map.page_mut() as *const u8).add(size_of::<LlmpPage>())
-                && buf_ptr <= (map.page_mut() as *const u8).add(map_size - size_of::<LlmpMsg>())
-            {
-                // The message header is in the page. Continue with checking the body.
-                let len = self.buf_len_padded as usize + size_of::<LlmpMsg>();
-                buf_ptr <= (map.page_mut() as *const u8).add(map_size - len)
-            } else {
-                false
-            }
+            buf_ptr > (map.page_mut() as *const u8).add(size_of::<LlmpPage>())
```

Comment:
@s1341 was this wrong previously?

Reply:
yes this is wrong
because it doesn't count the length of the payload

Reply:
the size of eop message is sizeof(LlmpMsg) + sizeof(LlmpPageSharedInfo) not sizeof(LlmpMsg)

before you didn't count sizeof(LlmpPageSharedInfo)

Reply:
Yes. This was wrong. @tokatoka and I debugged it together